src/plugins/crypto/crypto.c

```diff
@@ -220,7 +251,15 @@ void elektraCryptoHandleDestroy(elektraCryptoHandle *handle)
  */
 int elektraCryptoEncrypt(elektraCryptoHandle *handle, Key *k, Key *errorKey)
 {
+#ifdef ELEKTRA_CRYPTO_API_GCRYPT
 	return elektraCryptoGcryEncrypt(handle, k, errorKey);
+#endif
+
+#ifdef ELEKTRA_CRYPTO_API_OPENSSL
+	return elektraCryptoOpenSSLEncrypt(handle, k, errorKey);
+#endif
+
+	return (-1);
```

Comment:
strange "fallback", if you want to consider the case that neither ELEKTRA_CRYPTO_API_GCRYPT nor ELEKTRA_CRYPTO_API_OPENSSL are set, its better to give an `#error`.